shhext/service_test.go

```diff
@@ -0,0 +1,88 @@
+package shhext
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestShhExtSuite(t *testing.T) {
+	suite.Run(t, new(ShhExtSuite))
+}
+
+type ShhExtSuite struct {
+	suite.Suite
+
+	nodes    []*node.Node
+	services []*Service
+	whisper  []*whisper.Whisper
+}
+
+func (s *ShhExtSuite) SetupTest() {
+	s.nodes = make([]*node.Node, 2)
+	s.services = make([]*Service, 2)
+	s.whisper = make([]*whisper.Whisper, 2)
+	port := 32139
+	for i := range s.nodes {
+		i := i
+		key, err := crypto.GenerateKey()
+		s.NoError(err)
+		cfg := &node.Config{
+			Name: fmt.Sprintf("node-%d", i),
+			P2P: p2p.Config{
+				NoDiscovery: true,
+				MaxPeers:    20,
+				PrivateKey:  key,
+				ListenAddr:  fmt.Sprintf(":%d", port+i),
+			},
+		}
+		stack, err := node.New(cfg)
+		s.NoError(err)
+		s.whisper[i] = whisper.New(nil)
+		s.NoError(stack.Register(func(n *node.ServiceContext) (node.Service, error) {
+			return s.whisper[i], nil
+		}))
+		s.services[i] = New(nil)
+		s.NoError(stack.Register(func(n *node.ServiceContext) (node.Service, error) {
+			return s.services[i], nil
+		}))
+		s.Require().NoError(stack.Start())
+		s.nodes[i] = stack
+		s.services[i].ConnectWhisper(s.whisper[i])
+	}
+	s.nodes[0].Server().AddPeer(s.nodes[1].Server().Self())
+}
+
+func (s *ShhExtSuite) TestPostMessageWithConfirmation() {
+	confirmations := make(chan common.Hash, 1)
+	confirmationsHandler := func(hash common.Hash) {
+		confirmations <- hash
+	}
+	s.services[0].api.handler = confirmationsHandler
+	symID, err := s.whisper[0].GenerateSymKey()
+	s.NoError(err)
+	var hash common.Hash
+	client, err := s.nodes[0].Attach()
+	s.NoError(err)
+	s.NoError(client.Call(&hash, "shhext_post", whisper.NewMessage{
```

Comment:
@rasom how do you prefer to call this method? I wanted to make it clear that it is our extension and thus created a separate namespace